File: AccuPay/Domain/JobLevels/JobCategory.vb
Option Strict On

Imports System.Collections.ObjectModel
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.JobLevels

    <Table("jobcategory")>
    Public Class JobCategory

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property RowID As Integer?

        Public Property OrganizationID As Integer?

        Public Property Created As Date

        Public Property CreatedBy As Integer?

        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        Public Property Name As String

        Public Overridable Property JobLevels As ICollection(Of JobLevel)

        Public Sub New()
            JobLevels = New Collection(Of JobLevel)
            Created = Date.Now
        End Sub

    End Class

End Namespace

File: Accupay.Data/Entities/JobCategory.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Data.Entities
{
    [Table("jobcategory")]
    public class JobCategory
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? RowID { get; set; }

        public int? OrganizationID { get; set; }

        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        public string Name { get; set; }

        public virtual ICollection<JobLevel> JobLevels { get; set; }

        public JobCategory()
        {
            JobLevels = new Collection<JobLevel>();
            Created = DateTime.Now;
        }
    }
}